src/lib/mdns/Advertiser.h

```diff
@@ -179,34 +179,37 @@ class CommissionAdvertisingParameters : public BaseAdvertisingParams<CommissionA
     {
         if (deviceName.HasValue())
         {
-            strncpy(sDeviceName, deviceName.Value(), min(strlen(deviceName.Value()) + 1, sizeof(sDeviceName)));
-            mDeviceName = Optional<const char *>::Value(static_cast<const char *>(sDeviceName));
+            strncpy(mDeviceName, deviceName.Value(), min(strlen(deviceName.Value()) + 1, sizeof(mDeviceName)));
+            mDeviceNameHasValue = true;
         }
         return *this;
     }
-    Optional<const char *> GetDeviceName() const { return mDeviceName; }
+    bool HasValueDeviceName() const { return mDeviceNameHasValue; }
```

Comment:
you can keep the optional return for less code changes:

 Optional<const char *> GetDeviceName() const { 
    return mDeviceNameHasValue ? Optional::Value(mDeviceName) : Optional::Missing();
 }